jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -488,14 +488,24 @@ private void checkClassAccess(Class<?> targetClass) throws IllegalAccessExceptio
 				}
 				
 				int modifiers = targetClass.getModifiers();
-				if (Modifier.isPublic(modifiers)) {
-					/* Already determined that we have more than "no access" (public access) */
-					return;
-				} else if (Modifier.isProtected(modifiers)) {
-					/* Already determined that we have more than "no access" (public access) for a protected nested class */
+				
+				/* A protected class (must be a member class) is compiled to a public class as
+				 * the protected flag of this class doesn't exist on the VM level (there is no 
+				 * access flag in the binary form representing 'protected')
+				 */
+				if (Modifier.isProtected(modifiers)) {
+					/* Oracle Security-in-Depth Fix 8037326:
```

Comment:
@ChengJin01 Please remove this line.  It shouldn't be included in OpenJ9 code.

Reply:
Already removed.